OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameChatLogic.cs

```diff
@@ -265,7 +265,8 @@ public void AddNotificationWrapper(TextNotification notification)
 			if (!IsNotificationEligible(notification))
 				return;
 
-			chatOverlayDisplay?.AddNotification(notification);
+			if (!Game.Settings.Game.HideReplayChat || !world.IsReplay || notification.ClientId == TextNotificationsManager.ClientIdSystem)
```

Comment:
extract the conditions to a local function?

Reply:
extracted into `IsNotificationMuted`